stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -1586,7 +1586,7 @@ Base.reducedim_initarray0(A::SparseMatrixCSC, region, v0, ::Type{R}) where {R} =
     fill!(similar(dims->Array{R}(uninitialized, dims), Base.reduced_indices0(A,region)), v0)
 
 # General mapreduce
-function _mapreducezeros(f, op, ::Type{T}, nzeros::Int, v0) where T
+function _mapreducezeros(f, op, ::Type{T}, nzeros::U, v0) where {T, U<:Integer}
```

Comment:
Maybe a better solution is to convert `nzeros` to `Int` before calling these methods in order to avoid compiling a specialized version of this method.

Reply:
Thanks for the feedback @andreasnoack! I thought of that approach, but what would happen in that case for systems with a smaller `Int` than the type of `nzeros`?

eg: `Int` is `Int32` and `nzeros` has a value of `Int64` that overflows conversion to an `Int32`

Reply:
I'm pretty sure that this code is not even remotely robust in that case. E.g. https://github.com/JuliaLang/julia/blob/834cc44d9eb519f374dec1f38ca104e55f6bc9ca/stdlib/SparseArrays/src/sparsematrix.jl#L1610 since
```julia
julia> length(speye(50000))
-1794967296
```
Getting this to work reliably on 32 bit systems will require a bit more work. Hence I still think it is better to avoid the unnecessary method compilation by converting to `Int` before calling this function.